### PR TITLE
Update FriendsListScreen

### PR DESCRIPTION
**Pull request by nada-ezziti**
**Pull Request Description**

This pull request primarily focuses on updating the FriendListScreen composable to align with the new design provided in Figma and to enhance its functionality. Key updates include:

- **Updated Design**: Modified FriendListScreen to match the latest design specifications.

- **New Features**:
  - **Search Users by ID**: Added a search functionality to find users by their ID.
  - **Friend Requests**: Implemented the ability to send friend requests, as well as accept or decline incoming requests.
  - **Remove Friends**: Added the option to remove a friend from the list.
  
- **Code and Tests**:
  - Updated the getUserById method and its corresponding test.
  - Refactored FriendListScreen for simplified argument structure.
  - Added the saveUserToFireStore method to LoginScreen to handle saving user data to Firestore.
  - Achieved over 82% test coverage, ensuring all new functionalities are tested.

These changes enhance user interactions within the friends list and improve the screen’s design and functionality.

Capture d’écran 2024-10-28 à 11 11 25

### CODE REVIEW

File: app/src/androidTest/java/com/github/se/signify/ui/screens/profile/FriendsListScreenTest.kt [hunk 1]
```diff
@@ -3,34 +3,60 @@ package com.github.se.signify.ui.screens.profile
 import androidx.compose.runtime.mutableStateListOf
 import androidx.compose.ui.test.*
```
Comment (SaFouNaldo) [inline app/src/androidTest/java/com/github/se/signify/ui/screens/profile/FriendsListScreenTest.kt:4]: The tests are simple and efficient. Good job!

File: app/src/main/java/com/github/se/signify/MainActivity.kt [hunk 2]
```diff
@@ -110,17 +110,8 @@ fun SignifyAppPreview(context: Context, navigationState: MutableStateFlow<Naviga
             navigationActions = navigationActions)
       }
```
Comment (SaFouNaldo) [inline app/src/main/java/com/github/se/signify/MainActivity.kt:112]: the MainActivity is now clearer

**Review by SaFouNaldo — APPROVED** [review]
Nice work on the updates! 🎉

Noticed you’ve:

- Added extra tests for FriendsListScreen ✅
- Cleaned up MainActivity for better organization and readability 🧹
- Updated FriendsListScreen UI to align with the Figma design—thank you for improving the consistency! 🎨

**Review by samymessouak — APPROVED** [review]
I reviewed the changes to FriendListScreen closely and appreciate the well-thought-out improvements. The new features such as search by user ID, friend request handling, and friend removal significantly enhance user experience, and the UI matches the Figma design update perfectly.

Code Highlights:
The test cases for empty states and dynamic friend interactions, such as displayEmptyFriendsList() and testRemoveFriend(), are well-implemented to cover essential cases. The logic in FriendListScreen for dynamic UI responses in the Dialog and search bar also works seamlessly with the userViewModel.

Additionally, I ran the jacocoTestReport locally and confirmed the 80%+ test coverage—great job on the comprehensive test coverage! 🎉

Approving the PR! 🚀

**Pull request merged**